lunabot_nav/scripts/dstar_node.py

```diff
@@ -9,157 +9,224 @@
 
 from lunabot_nav.dstar import Dstar
 
-pose = []
-grid = []
-goal = []
-update_grid = False
-need_update_goal = False
-res = 0
-x_offset = 0
-y_offset = 0
+class DstarNode:
 
+    def __init__(self):
+        self.grid: np.ndarray[int] = np.array([]) # 2d array occupancy map: Occupancy probabilities from [0 to 100].  Unknown is -1.
+        self.pose: list[float] = [] # [x, y], in meters/odom frame
+        self.goal: list[float] = [] 
 
-def grid_subscriber(data):
-    global update_grid, grid, res, x_offset, y_offset
-    width = data.info.width
-    height = data.info.height
-    grid = np.reshape(data.data, (height, width))
-    res = data.info.resolution
-    x_offset = data.info.origin.position.x
-    y_offset = data.info.origin.position.y
-    update_grid = True
+        self.grid_update_needed: bool = False
+        self.goal_update_needed: bool = False
 
+        self.resolution: float = 0  # meters per grid cell
 
-def grid_update_subscriber(data):
-    global update_grid, grid, res, x_offset, y_offset
+        self.x_offset: float = 0 # real world pose of the point 0,0 in the grid
+        self.y_offset: float = 0
 
-    # grid = np.array(data.data).reshape(
-    #    (data.height, data.width), order="F"
-    # )
-    temp_map = grid.copy()
-    index = 0
-    for i in range(data.y, data.y + data.height):
-        for j in range(data.x, data.x + data.width):
-            temp_map[i][j] = data.data[index]
-            index += 1
+        self.dstar: Dstar = None
 
-    grid = temp_map.copy()
-    update_grid = True
+        self.path_sampling_rate = 5 # Take every <n-th> point from the path
```

Comment:
Lets make this a rosparam later (I personally don't like constants hidden inside files like this, I would at the very least move it to the top of the function)